ops/model.py

```diff
@@ -3640,12 +3642,31 @@ def secret_set(
         args = [id]
         if label is not None:
             args.extend(['--label', label])
+            self._secret_cache[id]['label'] = label
+        elif 'label' in self._secret_cache[id]:
+            args.extend(['--label', self._secret_cache[id]['label']])
         if description is not None:
             args.extend(['--description', description])
+            self._secret_cache[id]['description'] = description
+        elif 'description' in self._secret_cache[id]:
+            args.extend(['--description', self._secret_cache[id]['description']])
         if expire is not None:
             args.extend(['--expire', expire.isoformat()])
+            self._secret_cache[id]['expire'] = expire
+        elif 'expire' in self._secret_cache[id]:
+            args.extend(['--expire', self._secret_cache[id]['expire'].isoformat()])
         if rotate is not None:
             args += ['--rotate', rotate.value]
+            self._secret_cache[id]['rotate'] = rotate
+        elif 'rotate' in self._secret_cache[id]:
+            args += ['--rotate', self._secret_cache[id]['rotate'].value]
+        if content is None and 'content' in self._secret_cache[id]:
+            # Get the previous content from the unit agent's cache.
+            content = self.secret_get(id=id, peek=True)
+        elif content is not None:
+            # Don't store the actual secret content in memory, but put a sentinel there to indicate
+            # that the content has been set during this hook.
+            self._secret_cache[id]['content'] = object()
```

Comment:
This was a bit hard to understand... wdyt if the clauses were reordered?

```py
sec = self._secret_cache[id]  # debatable

if content is not None:
    sec["content-set"] = True
elif sec.get("content-set"):
    content = ...
```

Reply:
This does match the if statements above, so I'll adjust to match with that.

To me, it made more sense to have them in the order that things would happen, rather than the first clause being something that never happens before the second clause is triggered.